paddle/framework/tensor.h

```diff
@@ -51,27 +57,41 @@ class Tensor {
   struct Placeholder {
     virtual ~Placeholder() {}
     virtual void* Ptr() const = 0;
-    virtual Place Place() const = 0;
+    virtual paddle::platform::Place Place() const = 0;
     virtual size_t Size() const = 0;
   };
 
   template <typename T>
   struct PlaceholderImpl : public Placeholder {
-    PlaceholderImpl(Place pl, size_t size)
-        : ptr_(paddle::memory::Alloc(pl, size), paddle::memory::Deleter(pl)),
-          place_(pl),
+   private:
+    class Deleter {
```

Comment:
It seems that `Deleter` is pretty generally useful -- as long as memory blocks are allocated using `paddle::memory::Alloc`,  `Deleter` can free it.  How about we move `Deleter` to be `paddle::memory::Deleter`?